scripts/kernel-crawler.py

```diff
@@ -144,6 +144,21 @@
         # }
     ],
 
+    #
+    # Fedora Atomic repo is hard-coded to get the 4.17.x and 4.18.x (excluding rc) for now.
+    #
+    "Fedora-Atomic" : [
+        {
+            "root" : "https://kojipkgs.fedoraproject.org/packages/kernel/",
+            "version_discovery_pattern": "/html/body//a[regex:test(@href, '^4\.1[78].*/$')]/@href",
+            "build_discovery_pattern": "/html/body//a[regex:test(@href, '^[0-9]+\.[^r].*/$')]/@href",
```

Comment:
This works for 4.17/4.18, but atomic is tricky since they put all of the kernels up. Is it possible to look for kernels based on the date? For example, we could start with only kernels that were put out in 2018.

Is it possible to extend this to check the date associated with the href? For example, here's the html for https://kojipkgs.fedoraproject.org/packages/kernel/4.17.6/

```
<img src="/icons/folder.gif" alt="[DIR]"> <a href="100.fc27/">100.fc27/</a>               2018-07-12 00:13    -
<img src="/icons/folder.gif" alt="[DIR]"> <a href="200.fc28/">200.fc28/</a>               2018-07-12 00:29    -
```

It's not part of the href directly, so I don't know if xpath has a way to associate them.

If it's too much work, we can skip it for now.

Reply:
publish-date based discovery is one interesting option, as you had suggested.  we may consider white-list (based on user votes for instance), too. 4.17/18 are the latest kernels for fedora atomic; we'd like to use them as the initial offerings and see how customer would use them, and re-evaluate the discovery policy. I'll open a task for such re-evaluation.